### PR TITLE
Add support for conditional requests

### LINKED ISSUE #5: Speed up and reduce rate limit use
Currently when I want to update eg. you to my latest repos it will try to add you to all of my repos again. This is slow on my connection, which is not the fastest, and it can easily use up my rate limit.

Would be nice if this module could somehow make use of https://developer.github.com/v3/#conditional-requests

### PR DESCRIPTION
**Pull request by kevva**
Fixes #5.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,21 +1,37 @@
 'use strict';
 
+var Configstore = require('configstore');
 var githubRepos = require('github-repositories');
 var githubTokenUser = require('github-token-user');
 var ghGot = require('gh-got');
 var eachAsync = require('each-async');
+var conf = new Configstore('github-add-collab');
```
Comment (kevva) [inline index.js:8]: @sindresorhus, why does it create a new config each time?

File: index.js [hunk 2]
```diff
@@ -1,24 +1,37 @@
...
 
 function run(user, repos, opts, cb) {
 	eachAsync(repos, function (repo, i, next) {
+		var headers = {};
 		var url = 'repos/' + repo + '/collaborators/' + user;
+		var modified = conf.get(url);
```
Comment (kevva) [inline index.js]: Maybe set `repo/user` instead of `url`?

File: index.js [hunk 3]
```diff
@@ -1,24 +1,37 @@
...
-		}, function (err, data) {
+		}, function (err, data, res) {
 			if (err) {
 				cb(err);
 				return;
 			}
 
-			cb();
+			if (res.headers['Last-Modified'] !== modified) {
+				conf.set(repo + '/' + user, res.headers['Last-Modified']);
```
Comment (sindresorhus) [inline index.js]: Intentional?

``` js
console.log(repo + '/' + user);
//=> sindresorhus/got/kevva
```

File: index.js (hunk 1 above)
Comment (sindresorhus) [inline index.js:8]: Maybe related to https://github.com/kevva/github-add-collab/pull/6/files#r30256718 ? It doesn't for me.

File: index.js (hunk 3 above)
Comment (kevva) [inline index.js]: Whoops.

**Pull request merged**